docs/shared/deprecated/legacy-cache.md

```diff
@@ -0,0 +1,22 @@
+# Legacy Cache
+
+## tasksRunnerOptions
+
+As of Nx 20, the `tasksRunnerOptions` property in `nx.json` is deprecated. This property was used to register custom task runners. `tasksRunnerOptions` and custom task runners only work with the legacy file system cache.
+
+## NX_REJECT_UNKNOWN_LOCAL_CACHE
+
+The `NX_REJECT_UNKNOWN_LOCAL_CACHE` environment variable only works with the legacy file system local cache.
+
+## Next Steps
+
+In Nx 20, the local cache metadata and project graph are stored in a database, rather than using the file system. (Cache artifacts are still stored on the file system.) In Nx 21 the legacy file system cache will be removed and the database cache will be the only option. The database cache has two benefits:
+
+1. Cache reads and writes are faster.
+2. The local cache is more secure since other processes with access to the file system can no longer read or modify the cache.
+
+For most organizations, this feature is a net positive. If you are currently using a custom task runner or the `NX_REJECT_UNKNOWN_LOCAL_CACHE` environment variable, you are most likely providing your own custom [remote cache](/ci/features/remote-cache) storage location. You have several options moving forward:
```

Comment:
I think it would help people more  easily if we talk about the 3 scenarios separately.

1. Users who don't have `NX_REJECT_UNKNOWN_LOCAL_CACHE` set can move straight to  the db cache. They just need to remove the flag from `nx.json`.

2. Users who have `NX_REJECT_UNKNOWN_LOCAL_CACHE` have 3 options:
  a. Use Nx Cloud
  b. Purchase Powerpack and add the `@nx/powerpack/shared-fs-cache` package.
  c. Stop sharing cache via the fs
  
3. Users who are using a custom task runner have 3 different options from the people using `NX_REJECT_UNKNOWN_LOCAL_CACHE`
  a. Use Nx Cloud
  b. Look for a corresponding Powerpack cache. If one exists, purchase powerpack and download the corresponding package
  c. If there is no corresponding Powerpack cache and Nx Cloud does not offer the same features, reach out to us by creating an issue.
  d. Stop using the custom cache.
  
  @AgentEnder could we even point them to the specific section based on their runtime?